WooCommerce/Classes/POS/Controllers/PointOfSaleItemsController.swift

```diff
@@ -8,10 +8,20 @@ import protocol Yosemite.PointOfSalePurchasableItemFetchStrategy
 import enum Yosemite.PointOfSaleItemServiceError
 import struct Yosemite.POSVariableParentProduct
 import class Yosemite.Store
+import enum Yosemite.POSItemType
 
-enum ItemType {
+enum ItemListType {
     case products(search: Bool = false)
     case coupons
+
+    var itemType: POSItemType {
+        switch self {
+        case .coupons:
+            return .coupon
+        case .products(search: let search):
```

Comment:
As I understand, we would only return `.variation` here when we introduce search **within** the variations themselves. Right now, even if we pick a variation, we associate that search term with a product.

Reply:
Yeah, that's correct for current behaviour and probably correct for future searches within variations as well.

I added the `variation` case for consistency and fully modelling the item types. I'm still not sure whether it's actually correct for our use, just more correct than using the (now renamed) `ItemListType`. For example, it doesn't model any difference between simple products and variable parent products.